File: B13/Subhendu/Stacks/crawler-log-folder.py
class Solution:
    def minOperations(self, logs: List[str]) -> int:
        ### https://leetcode.com/problems/crawler-log-folder/

        stack = []

        for op in logs:
            if op == "../":
                if stack:
                    stack.pop()
            elif op == "./":
                pass
            else:
                stack.append(op)
        return len(stack)

File: B13/Hemanth/Stack/Crawler_Log_Folder.java
// https://leetcode.com/problems/crawler-log-folder/

class Solution {
    public int minOperations(String[] logs) {
        Stack<String> st = new Stack<>();
        for(String str:logs){
            if(!st.isEmpty() && str.equals("../")){
                st.pop();
            }
            else if(str.equals("./")){
                continue;
            }else if(!str.equals("../")){
                st.push(str);
            }
        }
        return st.size();
    }
}
